File: python/mongodb-driver/mongodb.py
import pymongo
from pymongo import MongoClient


def main():
    client = MongoClient('mongodb+srv://readonly:readonly@covid-19.hip2i.mongodb.net/test?retryWrites=true&w=majority')
    stats = client.get_database('covid19').get_collection('statistics')
    metadata = client.get_database('covid19').get_collection('metadata')

    print('Statistics for France, sorted by date descending, limit 20.')
    print_list(stats.find({'country': 'France', 'state': None}).sort('date', pymongo.DESCENDING).limit(15))

    print('\nMetadata')
    print_doc(metadata.find_one())

    print('\nIndex of the statistics collection.')
    print_list(stats.list_indexes())


def print_doc(doc):
    for k, v in doc.items():
        print(k, '=', v)


def print_list(docs):
    for d in docs:
        print(d)


if __name__ == '__main__':
    main()

File: nodejs/mongodb-driver/mongodb.js
const MongoClient = require("mongodb").MongoClient;

const uri =
  "mongodb+srv://readonly:readonly@covid-19.hip2i.mongodb.net/covid19";

const client = new MongoClient(uri, {
  useNewUrlParser: true,
  useUnifiedTopology: true,
});

client.connect((err) => {
  const covid19Database = client.db("covid19");
  const statistics = covid19Database.collection("statistics");
  const metadata = covid19Database.collection("metadata");

  // Query to get the last 5 entries for France (continent only)
  statistics
    .find({ country: "France" })
    .sort(["date", -1])
    .limit(15)
    .toArray((err, docs) => {
      if (err) {
        console.error(err);
      }
      console.log(docs);
    });

  //Query to get the last day data (limited to 5 docs here).
  metadata
    .find()
    .toArray((err, docs) => {
      if (err) {
        console.error(err);
      }
      const lastDate = docs[0].last_date;

      statistics
        .find({ date: { $eq: lastDate } })
        .limit(5)
        .toArray((err, docs) => {
          if (err) {
            console.error(err);
          }
          console.log(docs);
        });
    });

  // Query to get the last day data for all the countries within 500km of Paris.
  const lon = 2.341908;
  const lat = 48.860199;
  const earthRadius = 6371; // km
  const searchRadius = 500; // km

  metadata
    .find()
    .toArray((err, docs) => {
      if (err) {
        console.error(err);
      }
      const lastDate = docs[0].last_date;

      statistics
        .find({
          date: { $eq: lastDate },
          loc: {
            $geoWithin: {
              $centerSphere: [[lon, lat], searchRadius / earthRadius],
            },
          },
        })
        .limit(5)
        .toArray((err, docs) => {
          if (err) {
            console.error(err);
          }
          console.log(docs);
        });
    });
});
